plugins/httpdispatcher.koplugin/main.lua

```diff
@@ -0,0 +1,264 @@
+    end
+
+    local StreamMessageQueueServer = require("ui/message/streammessagequeueserver")
+    if self.http_socket == nil then
+        self.http_socket = StreamMessageQueueServer:new{
+            host = "*",
+            port = self.port,
+            receiveCallback = self:onRequest(),
+        }
+        self.http_socket:start()
+        self.http_messagequeue = UIManager:insertZMQ(self.http_socket)
+    end
+    logger.dbg("HttpDispatcher: Server listening on port " .. self.port)
+end
+
+function HttpDispatcher:stop()
+    logger.dbg("HttpDispatcher: Stopping server...")
+
+    if self.http_socket then
+        self.http_socket:stop()
+        self.http_socket = nil
+    end
+    if self.http_messagequeue then
+        UIManager:removeZMQ(self.http_messagequeue)
+        self.http_messagequeue = nil
+    end
+
+    -- Plug the hole in the Kindle's firewall
+    if Device:isKindle() then
+        os.execute(string.format("%s %s %s",
+            "iptables -D INPUT -p tcp --dport", self.port,
+            "-m conntrack --ctstate NEW,ESTABLISHED -j ACCEPT"))
+        os.execute(string.format("%s %s %s",
+            "iptables -D OUTPUT -p tcp --sport", self.port,
+            "-m conntrack --ctstate ESTABLISHED -j ACCEPT"))
+    end
+
+    logger.dbg("HttpDispatcher: Server stopped.")
+end
+
+function HttpDispatcher:onRequest(host, port)
+    -- NOTE: Closure trickery because we need a reference to *this* self *inside* the callback,
+    --       which will be called as a function from another object (namely, StreamMessageQueue).
+    local this = self
+    return function(data, id_frame)
+        local request = string.match(data, "[^\n]+") or ""
+        logger.dbg("HttpDispatcher: Received request: " .. request)
+        local params_string = request:match("%u+%s+%S+%?([^%s]+)%s+%S+") -- extract params
+        logger.dbg("HttpDispatcher: Params: ", params_string)
+
+        local params_arr = {}
+
+        if params_string then
+            for params in string.gmatch(params_string, "[^&]+") do
+                -- Split each parameter string into key and value using string.match
+                local key, value = string.match(params, "(.-)=(.*)")
+                if key and value then
+                    params_arr[key] = value
+                end
+            end
+        end
+
+        local response = "HTTP/1.1 200 OK" -- start of response header
+
+        if params_arr["event"] and params_arr["data"] then
+            if params_arr["datatype"] == "number" then
+                this:dispatchEvent(params_arr["event"], tonumber(params_arr["data"]))
+            else
+                this:dispatchEvent(params_arr["event"], params_arr["data"])
+            end
+        end
+
+        response = response .. "\r\n\r\n" -- end of response header
+
+        logger.dbg("HttpDispatcher: Sending response: " .. response)
+        this.http_socket:send(response, id_frame) -- send the response back to the client
```

Comment:
> Personally I don't need the server to host any http content since I'm directly sending commands from my remote, so still mulling over how to best implement that part.

I guess after you have parsed the query for the uri prefix you have specific handling for (ie. /event/ in my suggestion, alone for now), you would handle /mypage.html by just extracting `path = mypage.html`, and
```lua
local file = DataStorage:getDataDir() .. "/web/" .. path
if lfs.attributes(file, "mode") == "file" then
  -- read that file content and send it with your 200 OK
else
  -- send a 404
end
```
And if it is a directory, you could list its content and create and send a simple html listing of them.